src/Albany_StateManager.cpp

```diff
@@ -428,22 +428,204 @@ Albany::StateManager::registerSideSetStateVariable(
-    const std::string&                   meshPartName)
+    const std::string&                   meshPartName,
+    const bool                           useCollapsedLayout)
 
 {
-  return registerSideSetStateVariable(
-      sideSetName,
-      stateName,
-      fieldName,
-      dl,
-      ebName,
-      "",
-      0.0,
-      false,
-      outputToExodus,
-      "",
-      fieldEntity,
-      meshPartName);
+  if (useCollapsedLayout) {
+    return registerSideSetStateVariable_collapsed(
+        sideSetName,
+        stateName,
+        fieldName,
+        dl,
+        ebName,
+        "",
+        0.0,
+        false,
+        outputToExodus,
+        "",
+        fieldEntity,
+        meshPartName);
+  } else {
+    return registerSideSetStateVariable(
+        sideSetName,
+        stateName,
+        fieldName,
+        dl,
+        ebName,
+        "",
+        0.0,
+        false,
+        outputToExodus,
+        "",
+        fieldEntity,
+        meshPartName);
+  }
+}
+
+Teuchos::RCP<Teuchos::ParameterList>
+Albany::StateManager::registerSideSetStateVariable_collapsed(
+    const std::string&                   sideSetName,
+    const std::string&                   stateName,
+    const std::string&                   fieldName,
+    const Teuchos::RCP<PHX::DataLayout>& dl,
+    const std::string&                   ebName,
+    const std::string&                   init_type,
+    const double                         init_val,
+    const bool                           registerOldState,
+    const bool                           outputToExodus,
+    const std::string&                   responseIDtoRequire,
+    StateStruct::MeshFieldEntity const*  fieldEntity,
+    const std::string&                   meshPartName)
+{
+  TEUCHOS_TEST_FOR_EXCEPT(stateVarsAreAllocated);
+  using Albany::StateStruct;
+
+  // Create param list for SaveSideSetStateField evaluator
+  Teuchos::RCP<Teuchos::ParameterList> p =
+      Teuchos::rcp(new Teuchos::ParameterList(
+          "Save Side Set State " + stateName + " to/from Side Set Field " +
+          fieldName));
+  p->set<const std::string>("State Name", stateName);
+  p->set<const std::string>("Field Name", fieldName);
+  p->set<const std::string>("Side Set Name", sideSetName);
+  p->set<const Teuchos::RCP<PHX::DataLayout>>("Field Layout", dl);
+  p->set<const bool>("Use Collapsed Layout", true);
+
+  if (sideSetStatesToStore[sideSetName][ebName].find(stateName) !=
+      sideSetStatesToStore[sideSetName][ebName].end()) {
+    return p;  // Don't re-register the same state name
+  }
+
+  sideSetStatesToStore[sideSetName][ebName][stateName] = dl;
+
+  if (sideSetStateInfo.find(sideSetName) == sideSetStateInfo.end()) {
+    // It's the first time we register states on this side set, so we initiate
+    // the pointer
+    // TODO, when compiler allows, replace following with this for performance:
+    // sideSetStateInfo.emplace(sideSetName,Teuchos::rcp(new
```

Comment:
I think that now we can safely use emplace instead of insert

Reply:
Yes. Although that comment I put is a bit moot, since "performance" here is negligible. Still, using C++11 features is a good habit (and keeps the code a tad shorter).